### PR TITLE
Changed module dependency and usage field.

### PR DESCRIPTION
**Pull request by BasilSnowman**
Changed module dependency and usage field.

### CODE REVIEW

File: src/Menu/DomainMenuLinkTreeManipulators.php [hunk 1]
```diff
@@ -123,7 +124,7 @@ protected function menuLinkCheckAccess(MenuLinkInterface $instance, LanguageInte
 
       if (!$this->isAvailableOnAllAffiliates($entity)) {
```
Comment (andypost) [inline src/Menu/DomainMenuLinkTreeManipulators.php:125]: there's no "all" field in domain_entity, needs more refactoring

Comment (BasilSnowman) [reply src/Menu/DomainMenuLinkTreeManipulators.php:125]: fixed

Comment (BasilSnowman) [inline src/Menu/DomainMenuLinkTreeManipulators.php]: fixed

File: domain_menu_access.module [hunk 2]
```diff
@@ -59,19 +50,28 @@ function domain_menu_access_form_menu_link_content_form_alter(&$form, FormStateI
...
-          $definitions[$id]['class'] = DomainMenuAccessMenuBlock::class;
-          $definitions[$id]['provider'] = 'domain_menu_access';
+  // Check field storage and field definitions.
+  $field_storage = \Drupal::entityTypeManager()
+    ->getStorage('field_storage_config')
+    ->load('menu_link_content.' . DomainEntityMapper::FIELD_NAME);
+
+  $field = FieldConfig::loadByName('menu_link_content', 'menu_link_content', DomainEntityMapper::FIELD_NAME);
+
+  if ($field_storage && $field) {
```
Comment (andypost) [inline domain_menu_access.module]: field check is enough

File: src/Menu/DomainMenuLinkTreeManipulators.php [hunk 3]
```diff
@@ -146,13 +147,7 @@ protected function menuLinkCheckAccess(MenuLinkInterface $instance, LanguageInte
...
    */
   protected function isAvailableOnAllAffiliates(MenuLinkContent $entity) {
-    if ($entity->get(DOMAIN_ACCESS_ALL_FIELD)->isEmpty()) {
-      return FALSE;
-    }
-
-    $all_affiliates = $entity->get(DOMAIN_ACCESS_ALL_FIELD)->first()->getString();
-
-    return !empty($all_affiliates);
+    return empty($entity->get(DomainEntityMapper::FIELD_NAME)->isEmpty());
```
Comment (andypost) [inline src/Menu/DomainMenuLinkTreeManipulators.php:150]: `isEmpty()` is enough here
is available returns TRUE when this field is empty

**Pull request merged**